### ISSUE TITLE
Provide a simplified Markdown format for declaring Jupyter cells **with results**

### ISSUE THREAD

**Issue by grwlf** [jgm/pandoc#8598]
Currently, manual [describes](https://pandoc.org/MANUAL.html#jupyter-notebooks) requires users to specify divs (either fenced or native) around code sections in order for Pandoc to recognize them as code/result cells while converting Markdown documents to the Jupyter Notebooks format.

Unfortunately, 1) not all Markdown renderers support fenced divs 2) The document looks overly complex. 

**Let me suggest supporting a simplified mode as well, where result sections would just follow code sections immediately, possibly with some simple tag**

So, instead of:

```` markdown
:::::: {.cell .code execution_count=1}
```python
print("why so hard")
```
::: {.output .stream .stdout}
```result
why so hard
```
:::
::::::
````

we would have to write just

```` markdown
``` python
print("simple!")
```

``` result
simple!
```
````

Note, there are projects (e.g. [LitREPL](https://github.com/grwlf/litrepl.vim)) that support evaluation of Markdown cells as described.

**Comment by grwlf** [comment]
Related https://github.com/jgm/pandoc/issues/5992

**Comment by jgm** [comment]
Since I'm not a Jupyter user, I'd welcome more feedback from others on this proposal.
Obviously we don't want to make it impossible to have a python code example that isn't a code cell, but maybe the immediately following result cell is enough to make it unambiguous.

**Comment by alerque** [comment]
Would an attribute (say a `.cell` class) on the code block be serviceable here?

~~~markdown
``` python {.cell}
print("with attrs")
```
``` {.result}
with attrs
```
~~~

**Comment by jgm** [comment]
That might be incompatible with the goal of making the syntax available to non-pandoc markdown variants.

**Comment by grwlf** [comment]
I have sketched the following [test.md](https://gist.github.com/grwlf/c0b369f1b15d609575ba5f0f1ca3250f)  which uses 3 types of cell declaration syntax:
1. The one suggested at the top of this PR
2. The `python { .cell }` syntax suggested by @alerque 
3. Additionally, the `{.python .cell }` syntax which resembles the syntax which is supported by another popular tool [codebraid](https://codebraid.org/getting_started/)

Then I tried to display the file using the following renderers:
1. Online Github renderer
2. Chromium with the [Markdown Viewer plugin](https://chrome.google.com/webstore/detail/markdown-viewer/ckkdlimhmcjmikdlpkmbgfkaikojcbjk) (`chromium test.md`)
3. Pandoc-to-HTML converter (`pandoc test.md  --metadata title=test -s -t slidy  -o test.html ; firefox test.html `). I used Pandoc version `2.17.1.1`

Here is the results (C - Correct, H - Python code is highlighted, B - Bad: non-code markup is visible) :

| Syntax | Github online | Chromium-based | Pandoc-to-HTML |
|---|---|---|---|
| V1 | C,H | C,H | C,H |
| V2 | C,H | B | B |
| V3 | C | C |  C,H |

So I would suggest keep the originally proposed syntax, or, as a second option, think about supporting the Codebraid version (` {.python .cell}` ) rather than `python { .cell }`.

**Comment by jgm** [comment]
@fperez any thoughts on this?

**Comment by fperez** [comment]
My first thought is that the concept of output in Jupyter goes well beyond plaintext. Our output objects are "mime bundles" (JSON dicts keyed by mimetype and with base64 encoded values if necessary). You can see e.g. [R code](https://cran.r-project.org/web/packages/IRdisplay/readme/README.html) that supports these.

I'm not right now deep in the weeds of our format machinery, but this is the kind of thing that would make for good discussion at the upcoming [Notebook format workshop](https://blog.jupyter.org/jupyter-community-workshop-the-notebook-file-format-8133ed606118), if nothing else in order to have an informed conversation between Jupyter devs and the Pandoc team.

Pinging @rowanc1 @stevejpurves @agoose77 in case any of them has further thoughts as our MyST work progresses.

**Comment by jgm** [comment]
It's a good point that we cannot assume that the output will be representable by a code block.  But I wonder if that's a problem with this proposal.  Pandoc already has ways of representing arbitrary code and output cells in its markdown format. The proposal here is to provide an *alternative, simplified* markdown syntax in addition to the current, more verbose markdown syntax (which can represent arbitrary code and output cells).  If the alternative syntax can only represent simple output cells, that isn't necessarily a problem.

**Comment by rowanc1** [comment]
In my experience with Juptyer most of the time outputs are (1) rich mime-bundles (images, interactive figures, latex, ansi, maps, etc.); and (2) a list (i.e. multiple outputs to a single code execution). When people represent notebooks as markdown (e.g. in [MyST](https://jupyterbook.org/en/stable/file-types/myst-notebooks.html) or with jupytext or with quarto/rmd) the outputs are _not_ stored because of this complexity which is usually at odds with a human-written format like md.

I suppose because of that I am having a hard time wrapping my head around the use case here, but if it is just an optional simplification of what already exists and make it work in more renderers: 👍

**Comment by grwlf** [comment]
> I suppose because of that I am having a hard time wrapping my head around the use case here, but if it is just an optional simplification of what already exists and make it work in more renderers: +1

This suggestion is primarily intended to support the text-mode software development. Tools like Codebraid or LitREPL are designed with this goal in mind and they could be used as a limited but lightweight Jupyter alternative. In this scenario, conversion to the Jupyter format is required for sharing sketches or demonstrations with external teams at the end of development cycles.

Of cause in text mode we don't aim to render things like interactive figures and may assume only a limited support of the rich-text outputs. Note, however, that while this is probably beyond the scope of discussion, there are terminals which can display images even in text mode. For example, this [fork of st](https://github.com/sergei-grechanik/st) could display a picture sent from a tmux-session running on a remote server over SSH. Consider also this [Vim plugin project](https://github.com/sergei-grechanik/vim-terminal-images) as a working poof of concept.

**Comment by fperez** [comment]
I agree with @rowanc1 - if it helps simplify existing worfklows, I don't see any reason why not. And you could imagine, with some good tooling support, being able to also store rich outputs in a sidecar file with convenient linkage, so the main doc remains plaintext while binary results are stored in the sidecar. Something along the lines of

<pre>

```python
f(x)
```
```result
::sidecar:my-uuid.png
```
</pre>

or similar. Lots of details would need to be worked out for something like this, but it would allow a human, terminal-based workflow to still represent richer documents via plaintext.

Again - these are some of the conversations that I'm sure will take place at the notebook format workshop, I hope some of you have a chance to participate (sadly I can't).
